src/grpc/grpc_server.cc

```diff
@@ -2435,6 +2435,104 @@ Server::Create(
   return nullptr;  // success
 }
 
+TRITONSERVER_Error*
+Server::Create(
+    std::shared_ptr<TRITONSERVER_Server>& server, UnorderedMapType& data,
+    const RestrictedFeatures& restricted_features,
+    std::unique_ptr<Server>* service)
+{
+  Options grpc_options;
+
+  GetOptions(grpc_options, data);
```

Comment:
The consistency of your API is an issue. In `GetOptions` we can be handling a `throw` statement but you do no try-catch to handle the error cases. Also, in the next line you handle a TRITONSERVER_Error. If we are to be consistent in our API design, one of two things should happen:
1. GetOptions returns a TRITONSERVER_Error
2. Create throws instead of returning a TRITONSERVER_Error.

The I prefer the first option because it informs the user of the API to check for error cases where as implicitly throwing doesn't tell the user if the function returns an error, if anything at all.

Reply:
Thank you for the suggestion! I have re-factored the code to have the following: `get_value(...)` returns `TRITONSERVER_Error*`. And this error message is propagated further to the helper functions: `GetSocketOptions()`, `GetSslOptions()`, `GetKeepAliveOptions()`, and `GetOptions()`. These functions are wrapped with `RETURN_IF_ERR` macro defined in `common.h`.